From c2c1012ae860eb1099faa3c036ee1b65d541f0f1 Mon Sep 17 00:00:00 2001
From: Josef Schlehofer <pepe.schlehofer@gmail.com>
Date: Sat, 9 Nov 2019 10:14:58 +0100
Subject: [PATCH] strace: remove patch which fixes build on powerpc

This patch, which was added in 2016, is no longer needed. Actually,
nowadays in branch openwrt-19.07 and master it breaks build for powerpc.

With this patch, the compilation of strace fails:

In file included from syscall.c:1002:
syscall.c: In function 'get_instruction_pointer':
./linux/powerpc/arch_regs.c:11:29: error: invalid use of undefined type 'struct pt_regs'
 #define ARCH_PC_REG ppc_regs.nip
                             ^
syscall.c:1246:25: note: in expansion of macro 'ARCH_PC_REG'
  *ip = (kernel_ulong_t) ARCH_PC_REG;

In file included from syscall.c:1002:
syscall.c: At top level:
./linux/powerpc/arch_regs.c:8:23: error: storage size of 'ppc_regs' isn't known
 static struct pt_regs ppc_regs;
                       ^~~~~~~~

Fixes: FS#2582 ("PowerPC8540: Strace error: invalid use of undefined type 'struct pt_regs'"

Signed-off-by: Josef Schlehofer <pepe.schlehofer@gmail.com>
---
 .../100-workaround--pt-reg-collisions-ppc.patch | 17 -----------------
 1 file changed, 17 deletions(-)
 delete mode 100644 package/devel/strace/patches/100-workaround--pt-reg-collisions-ppc.patch

diff --git a/package/devel/strace/patches/100-workaround--pt-reg-collisions-ppc.patch b/package/devel/strace/patches/100-workaround--pt-reg-collisions-ppc.patch
deleted file mode 100644
index 93e2980749..0000000000
--- a/package/devel/strace/patches/100-workaround--pt-reg-collisions-ppc.patch
+++ /dev/null
@@ -1,17 +0,0 @@
---- a/ptrace.h
-+++ b/ptrace.h
-@@ -30,7 +30,14 @@
- #  define ptrace_peeksiginfo_args XXX_ptrace_peeksiginfo_args
- # endif
- 
-+#if POWERPC
-+#include <linux/types.h>
-+#define __ASSEMBLY__
-+#endif
- # include <linux/ptrace.h>
-+#if POWERPC
-+#undef __ASSEMBLY__
-+#endif
- 
- # ifdef HAVE_STRUCT_IA64_FPREG
- #  undef ia64_fpreg
-- 
2.20.1

